src/test/java/com/feedzai/commons/sql/abstraction/engine/impl/abs/AbstractEngineSchemaTest.java

```diff
@@ -188,6 +194,59 @@ public void udfTimesTwoTest() throws Exception {
         }
     }
 
+    /**
+     * Tests that the different DB engines respect the fetch size set in a query.
+     *
+     * @throws Exception If any error occurs, thus failing the test.
+     */
+    @Test
+    public void testFetchSize() throws Exception {
+        final DatabaseEngineDriver engineDriver = DatabaseEngineDriver.fromEngine(properties.getProperty(ENGINE));
+        final ExecutorService executor = Executors.newCachedThreadPool();
+
+        final TestRouter testRouter = new TestRouter(executor, engineDriver.defaultPort());
+        testRouter.init();
+
+        final Properties testProps = TestRouter.getPatchedDbProperties(properties, testRouter.getDbPort(), testRouter.getLocalPort());
+        testProps.setProperty(PdbProperties.SOCKET_TIMEOUT, "2");
+        testProps.setProperty(PdbProperties.CHECK_CONNECTION_TIMEOUT, "2");
+
+        try (final DatabaseEngine engine = DatabaseFactory.getConnection(testProps)) {
+            final DbEntity entity = createSpecialValuesEntity();
+            engine.addEntity(entity);
+            final String entityName = entity.getName();
+
+            for (int i = 0; i < 10; i++) {
+                engine.persist(entityName, entry().set(ID_COL, i).build());
+            }
+
+            // besides the connection timeouts, also set 2 seconds timeout for the query; fetch size = 3
+            try (ResultIterator iterator = engine.iterator(select(all()).from(table(entityName)), 3, 2)) {
+                // if the first 3 rows aren't fetched immediately on the query, they will be fetched next
+                iterator.next();
+
+                // break the connection to the DB
+                testRouter.breakConnections();
+
+                // get next result, which should already have been fetched; connection shouldn't be needed and this shouldn't fail
+                assertThatCode(iterator::next)
+                        .doesNotThrowAnyException();
+
+                // same as previous; ensures that the 3rd result is already fetched, and retries don't prevent from getting it
+                assertThatCode(iterator::next)
+                        .doesNotThrowAnyException();
+
+                // 4th row should be fetched now, but since the connection is broken it should fail
+                // (unless the driver already fetched all results, which should fail the test)
+                assertThatThrownBy(iterator::next)
+                        .isInstanceOf(Exception.class);
```

Comment:
Shouldn't this check the specific exception that is thrown when it can't be fetched from the DB?